roles/job/templates/job_definition.yml.j2

```
@@ -15,9 +15,15 @@ spec:
           imagePullPolicy: Always
           env:
             - name: TOWER_OAUTH_TOKEN
-              value: "{{ tower_config_secret['resources'][0]['data']['token'] | b64decode }}"
+              valueFrom:
+                secretKeyRef:
+                  name: "{{ tower_config_secret['resources'][0]['metadata']['name'] }}"
```

I think we can directly use `tower_auth_secret` for this instead of using the resource name.
```suggestion
                  name: "{{ tower_auth_secret }}"
```